client/handle_http.go

```diff
@@ -2256,6 +2262,73 @@ func runPut(request *http.Request, responseChan chan<- *http.Response, errorChan
 
 }
 
+// This function queries the director with a PROPFIND to attempt to get the 'collections url'. If a propfind is not allowed on the director
+// We fall back to the deprecated dirlisthost in the namespace
+func getCollectionsUrl(ctx context.Context, remoteObjectUrl *url.URL, namespace namespaces.Namespace, directorUrl string) (collectionsUrl *url.URL, err error) {
```

Comment:
Note: this function is taken from this PR: https://github.com/PelicanPlatform/pelican/pull/1257

So that PR should most likely get approved and merged first but I don't think it should matter.